evm/p2p/test_auth.py

```diff
@@ -101,6 +111,35 @@ def test_handshake():
     assert initiator_ingress_mac.digest() == test_values['initial_ingress_MAC']
     assert initiator_egress_mac.digest() == test_values['initial_egress_MAC']
 
+    # Finally, check that two Peers configured with the secrets generated above understand each
+    # other.
+    responder_reader = asyncio.StreamReader()
+    responder_writer = None
+    initiator_reader = None
+    # Link the initiator's writer to the responder's reader.
+    initiator_writer = type(
+        "mock-streamwriter",
+        (object,),
+        {"write": lambda data: responder_reader.feed_data(data)}
+    )
+    initiator_peer = Peer(
+        remote=initiator.remote, privkey=initiator.privkey, reader=initiator_reader,
+        writer=initiator_writer, aes_secret=initiator_aes_secret, mac_secret=initiator_mac_secret,
+        egress_mac=initiator_egress_mac, ingress_mac=initiator_ingress_mac)
+    responder_peer = Peer(
+        remote=responder.remote, privkey=responder.privkey, reader=responder_reader,
+        writer=responder_writer, aes_secret=aes_secret, mac_secret=mac_secret,
+        egress_mac=egress_mac, ingress_mac=ingress_mac)
+
+    # The hello msg sent by the initiator is going to be fed directly into the responder's
+    # reader, and thus read_msg() will return immediately.
+    initiator_peer.send_hello()
+    msg = yield from responder_peer.read_msg()
+
+    cmd_id = rlp.decode(msg[:1], sedes=sedes.big_endian_int)
+    assert cmd_id == Hello.id
+    responder_peer.process_msg(msg)
```

Comment:
feels odd to have the test end with something other than an assertion... is there something *after* this that could be asserted or a reason that the test ends with this statement?

Reply:
Right now there's nothing to assert after that; it's only checking that the message processing doesn't raise an exception. In the branch I'm currently working on I assert that the peer has been configured with sub-protocols matching the remote's capabilities, though